### PR TITLE
Add support for pitch with new audio engine playback

### PR DESCRIPTION
**Pull request by stevetranby**
Only tested on Mac so far, so need to test other platforms. Tizen (and any others) assert "not supported".

### CODE REVIEW

File: cocos/audio/android/AudioEngine-inl.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/****************************************************************************
+s/****************************************************************************
```
Comment (minggo) [inline cocos/audio/android/AudioEngine-inl.h:1]: why add `+s`?

File: cocos/audio/winrt/AudioCachePlayer.cpp [hunk 2]
```diff
@@ -337,6 +337,22 @@ void AudioPlayer::setVolume(float volume)
     }
 }
 
+void AudioPlayer::setPitch(float pitch)
+{
+	CCASSERT("AudioPlayer::setPitch not implemented yet in cache player!");
```
Comment (minggo) [inline cocos/audio/winrt/AudioCachePlayer.cpp:342]: This feature is not implemented but why there are codes below?

File: tests/cpp-tests/Classes/BillBoardTest/BillBoardTest.cpp [hunk 3]
```diff
@@ -239,6 +239,8 @@ void BillBoardTest::addNewBillBoradWithCoords(Vec3 p)
     {
         auto billborad = BillBoard::create(imgs[(unsigned int)(CCRANDOM_0_1() * 1 + 0.5)]);
         billborad->setScale(0.5f);
+        //billborad->setFlippedX(true);
+        billborad->setRotation3D(Vec3(0,180,0));
```
Comment (minggo) [inline tests/cpp-tests/Classes/BillBoardTest/BillBoardTest.cpp:243]: I think these modifications are nothing about this PR.

File: tests/cpp-tests/Classes/UITest/CocoStudioGUITest/UIListViewTest/UIListViewTest.cpp [hunk 4]
```diff
@@ -853,6 +853,12 @@ bool UIListViewTest_Magnetic::init()
             eNextType = ListView::MagneticType::NONE;
             sString = "NONE";
         }
+        else
+        {
+            eNextType = ListView::MagneticType::NONE;
+            sString = "NONE";
+        }
```
Comment (minggo) [inline tests/cpp-tests/Classes/UITest/CocoStudioGUITest/UIListViewTest/UIListViewTest.cpp:860]: Nothing about this PR?

File: tests/cpp-tests/Classes/UnitTest/RefPtrTest.cpp [hunk 5]
```diff
@@ -313,7 +313,7 @@ void RefPtrTest::onEnter()
         
         // Note: std::move will turn the rvalue into an rvalue reference and thus cause the move constructor to be invoked.
         // Have to use this because the compiler will try and optimize how we handle the return value otherwise and skip the move constructor.
-        RefPtr<__String> theString(std::move(someFunc()));
+        RefPtr<__String> theString(someFunc());
```
Comment (minggo) [inline tests/cpp-tests/Classes/UnitTest/RefPtrTest.cpp:316]: nothing about this PR?

File: tests/cpp-tests/Classes/UnitTest/UnitTest.cpp [hunk 6]
```diff
@@ -517,7 +517,7 @@ void TemplateMapTest::onEnter()
         CCASSERT(e.second->getReferenceCount() == 2, "e.second's reference count is 2.");
     }
 
-    mapForSelfAssign = std::move(mapForSelfAssign);
+    //mapForSelfAssign = std::move(mapForSelfAssign);
```
Comment (minggo) [inline tests/cpp-tests/Classes/UnitTest/UnitTest.cpp:520]: all the modifications are nothing about this PR

File: cocos/audio/include/AudioEngine.h [hunk 7]
```diff
@@ -113,60 +113,77 @@ class EXPORT_DLL AudioEngine
...
      * @param volume Volume value (range from 0.0 to 1.0).
      */
     static void setVolume(int audioID, float volume);
 
-    /** 
+    /**
+     * Sets pitch for an audio instance.
+     *
+     * @param audioID An audioID returned by the play2d function.
+     * @param pitch Pitch value, 0.5 = half pitch, 2.0 = double pitch, etc default = 1.0.
```
Comment (minggo) [inline cocos/audio/include/AudioEngine.h:166]: `default = 1.0` does it mean if we don't set pitch, the default is 1.0 set by the player itself as i don't see there is any codes to set the pitch.

Comment (stevetranby) [reply cocos/audio/include/AudioEngine.h:166]: Default is unset or 1.0 yes.

**Comment by stevetranby** [comment]
Looks like I must've got a merge from my v3 branch mixed in. I'll delete this and the one or two other work in progress PR's. I pushed them up here to reference in some discussions so others could see at least some work on it. I'll keep these over in my fork only until fully ready going forward. Sorry to take up your time reviewing this one.

**Pull request closed without merging**